matRad_rayTracing.m

```diff
@@ -126,16 +126,29 @@
 
         for j = 1:ct.numOfCtScen
             % calc radiological depths
-
-            % eq 14
-            % It multiply voxel intersections with \rho values.
-            d = l .* rho{j}; %Note. It is not a number "one"; it is the letter "l"
-
-            % Calculate accumulated d sum.
-            dCum = cumsum(d)-d/2;
-
-            % write radiological depth for voxel which we want to remember
-            radDepthCube{j}(ixHitVoxel(ixRememberFromCurrTracing)) = dCum(ixRememberFromCurrTracing);
+            
+            % calculate radiological depth only inside the patient in an
+            % area with boundaries above the skin threshold
+            aboveThreshold = find(rho{j} > skinThreshold);
+            if numel(aboveThreshold) > 0
+                start = aboveThreshold(1);
+                stop = aboveThreshold(end);
+                
+                rho{j} = rho{j}(start:stop);
+                l = l(start:stop);
+                ixHitVoxel = ixHitVoxel(start:stop);
+                ixRememberFromCurrTracing = ixRememberFromCurrTracing(start:stop);
+
```

Comment:
I have to change the assignment of the new indices, this will fail for skinTreshhold > 0 in combination with multiple scenarios.
I don't really like the 4 new assignments I make here, but this was the best way I could think of at that point.
Do you have a better idea how to do this?